ansible/library/test_facts.py

```diff
@@ -103,28 +104,22 @@ class ParseTestbedTopoinfo():
     '''
     def __init__(self, testbed_file):
         self.testbed_filename = testbed_file
-        self.testbed_topo = {}
+        self.testbed_topo = defaultdict()
 
     def read_testbed_topo(self):
+        CSV_FIELDS = ('conf-name', 'group-name', 'topo', 'ptf_image_name', 'ptf', 'ptf_ip', 'server', 'vm_base', 'dut', 'comment')
```

Comment:
Just wondering if we would want to allow different field orderings. If so, we could make the first line of the file the "header" line, remove the leading `#` and parse the first line separately. We could even add a check to ensure all the expected fields are present. Maybe this could be an enhancement; a separate, future PR?

Reply:
done in the latest iteration

Reply:
Thanks. But I was actually referring to possibly removing the leading `#` from the line in the actual CSV file. Then when we parse the file here, we first parse the first line ("header" line) only, which will give us the fields in the order they were supplied, then we can parse the file using the order of the fields supplied. There is no rush on this feature, just a thought for a future PR which could add some flexibility.

Reply:
This is a great idea. We had similar code in syncd when passing the port_config.ini (in c++). I think we could use this good feature.

However, if this change is to address some immediate test issue. Please come back with another PR for the enhancement.

Reply:
DictReader does this if the fieldsname param is not provided. It will assume the first line to be a header line that provides fields name ('#' does not need to be present if fieldsname param is provided.) The benefits here is that we know the order and so will be able to check for commented lines.